src/main/java/com/robertx22/mine_and_slash/a_libraries/dmg_number_particle/particle/InteractionResultHandler.java

```diff
@@ -0,0 +1,121 @@
+package com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle;
+
+import com.google.common.collect.ImmutableMap;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.impl.DamageNullifiedParticle;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.impl.ElementDamageParticle;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.impl.HealParticle;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.style.Original;
+import com.robertx22.mine_and_slash.a_libraries.dmg_number_particle.particle.style.Row;
+import com.robertx22.mine_and_slash.config.forge.ClientConfigs;
+import com.robertx22.mine_and_slash.uncommon.enumclasses.Elements;
+import com.robertx22.mine_and_slash.uncommon.utilityclasses.ClientOnly;
+import com.robertx22.mine_and_slash.uncommon.utilityclasses.NumberUtils;
+import com.robertx22.mine_and_slash.vanilla_mc.packets.interaction.IParticleSpawnMaterial;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.Minecraft;
+import net.minecraft.sounds.SoundSource;
+import net.minecraft.world.entity.Entity;
+
+import java.util.Comparator;
+import java.util.Map;
+import java.util.Random;
+import java.util.function.BiConsumer;
+
+public class InteractionResultHandler {
+
+
+
+    public enum ParticleSpawnType {
+        DAMAGE(ClientConfigs.getConfig().DAMAGE_PARTICLE_STYLE.get().damageStrategy, new ElementDamageParticle.DamageInformation(null, null, false)),
+        NULLIFIED_DAMAGE(ClientConfigs.getConfig().DAMAGE_PARTICLE_STYLE.get().nullifiedDamageStrategy, DamageNullifiedParticle.Type.DODGE),
```

Comment:
this part seems bad. Do not save config values and especially dont call them in constructors